client/client-lib/src/lib.rs

```diff
@@ -759,23 +782,29 @@ impl Client<UserClientConfig> {
         let duration_since_epoch =
             Duration::from_secs_f64(js_sys::Date::new_0().get_time() / 1000.);
 
-        let invoice = InvoiceBuilder::new(network_to_currency(self.config.0.wallet.network))
-            .amount_milli_satoshis(amount.milli_sat)
-            .description(description)
-            .payment_hash(payment_hash)
-            .payment_secret(payment_secret)
-            .duration_since_epoch(duration_since_epoch)
-            .min_final_cltv_expiry(18)
-            .payee_pub_key(node_public_key)
-            .private_route(gateway_route_hint)
-            .expiry_time(Duration::from_secs(
-                expiry_time.unwrap_or(DEFAULT_EXPIRY_TIME),
-            ))
-            .build_signed(|hash| {
-                self.context
-                    .secp
-                    .sign_ecdsa_recoverable(hash, &node_secret_key)
-            })?;
+        let invoice = InvoiceBuilder::new(network_to_currency(
+            self.config
+                .0
+                .get_module::<WalletClientConfig>("wallet")
+                .expect("must have wallet config available")
+                .network,
+        ))
```

Comment:
Good enough for now, but oh boy do I hate parsing the cfg over and over again here.

Reply:
This logic will either live inside client side of the module that will have the client config typed at all times, or the end-client-binary should know which modules it needs and store them typed somewhere (parse once).